src/coreclr/nativeaot/Runtime/inc/CommonTypes.h

```diff
@@ -61,4 +61,12 @@ typedef struct _GUID {
 } GUID;
 #endif // FEATURE_EVENT_TRACE && !_INC_WINDOWS
 
+// Hijack funcs are not called, they are "returned to". And when done, they return to the actual caller.
+// Thus they cannot have any parameters or return anything.
+#if defined(HOST_X86) && !defined(HOST_UNIX)
+typedef void __fastcall HijackFunc();
+#else
+typedef void HijackFunc();
+#endif
```

Comment:
```suggestion
typedef void HijackFunc();
```
We should not need __fastcall on the hijack func. Can we update the asm mangled name instead to avoid the ifdef?